src/Breadcrumb/Breadcrumb.tsx

```diff
@@ -1,32 +1,46 @@
-import * as React from "react";
-import "./breadcrumb-style.scss";
+import React from "react";
+import { isPrimitive } from "@sebgroup/frontend-tools/dist/isPrimitive";
+import "./breadcrumb.scss";
 
 export interface BreadcrumbProps {
+    /** Element class name */
     className?: string;
+    /** Element id */
     id?: string;
-    list: Array<string | React.ReactNode>;
-    onClick?: (i: number, e?: React.MouseEvent<HTMLLIElement>) => void;
+    /** The list of breadcrumb items */
+    list: Array<BreadcrumbItem>;
+    /** onClick callback */
+    onClick?: (e?: React.MouseEvent<HTMLAnchorElement>) => void;
 }
 
-export const Breadcrumb: React.FunctionComponent<BreadcrumbProps> = React.memo(
-    (props: BreadcrumbProps): React.ReactElement<void> => {
-        function onClick(index: number, e?: React.MouseEvent<HTMLLIElement>): void {
-            if (index !== props.list.length - 1 && props.onClick) {
-                props.onClick(index, e);
-            }
-        }
-        return (
-            <div className={"custom-breadcrumb" + (props.className ? ` ${props.className}` : "")} id={props.id}>
-                <nav aria-label="breadcrumb">
-                    <ol className="breadcrumb">
-                        {props.list.map((item: string | React.ReactNode, i: number) => (
-                            <li key={i} className={"breadcrumb-item" + (i === props.list.length - 1 ? " active" : "")} onClick={(e: React.MouseEvent<HTMLLIElement>) => onClick(i, e)}>
-                                {item}
-                            </li>
-                        ))}
-                    </ol>
-                </nav>
-            </div>
-        );
-    }
+export interface BreadcrumbItem {
+    /** The content to be displayed in each breadcrumb item */
+    text: React.ReactNode;
+    /**
+     * The link to where it leats. This is used to enable openning the link in new tab.
+     * Additionally, you can access it in the event passed with the onClick callback
+     */
+    href?: string;
+    /** The title of the anchor tag, used for accessibility to describte where the link takes you */
+    title?: string;
+}
+
+export const Breadcrumb: React.FC<BreadcrumbProps> = React.memo(
+    (props: BreadcrumbProps): React.ReactElement<void> => (
+        <nav aria-label="breadcrumb" className={props.className} id={props.id}>
+            <ol className="seb breadcrumb">
+                {props.list.map((item: BreadcrumbItem, i: number) => {
+                    const isLast: boolean = i === props.list.length - 1;
+
+                    return (
+                        <li key={i} className={"breadcrumb-item" + (isLast ? " active" : "")} aria-current={isLast ? "page" : null}>
```

Comment:
Wouldn't this result in a rect key duplicate in an extreme cases ? . I suggest using or appending some variating text to the key.

Reply:
I've read about it and even tried it. The problem is that I cannot create random keys since that would cause the whole list to re-render every time, and I don't have a way of appending the index since all the information passed from the user might be of type `ReactNode`. However, this is not a scenario that raises concerns, simply because this is not a dynamic list where you insert items in between. If the user chooses to do that, React will detect a change in props and cause a render with new keys .

Reply:
My concern is not the fact that user can modify the data, my concern is a situation where by user renders multiple instances of the component in the sate UI . In my experience, when in complex or redundant usage, this kind of arrangement tends to throw the duplicate keys errors.

Reply:
That wouldn't be an issue, the key is only used by the host component